test/unittests/test_qfr_functionality.cpp

```diff
@@ -1937,3 +1937,133 @@ TEST_F(QFRFunctionality, dumpAndImportTeleportation) {
   ASSERT_EQ(qcImported.size(), 1);
   EXPECT_EQ(qcImported.at(0)->getType(), OpType::Teleportation);
 }
+
+TEST_F(QFRFunctionality, addControlStandardOperation) {
+  auto op = StandardOperation(2, Targets{0}, OpType::X);
+
+  op.addControl(Control{1, Control::Type::Pos});
+  op.addControl(Control{2, Control::Type::Pos});
+
+  ASSERT_EQ(op.getNcontrols(), 2);
+  auto expectedControls =
+      Controls{Control{1, Control::Type::Pos}, Control{2, Control::Type::Pos}};
+  EXPECT_EQ(op.getControls(), expectedControls);
+  op.removeControl(Control{1, Control::Type::Pos});
+  auto expectedControlsAfterRemove = Controls{Control{2, Control::Type::Pos}};
+  EXPECT_EQ(op.getControls(), expectedControlsAfterRemove);
+  op.clearControls();
+  EXPECT_EQ(op.getNcontrols(), 0);
+}
```

Comment:
This test (and also the ones below) can be simplified a little to be less verbose.
Also, if you want to add 2 controls to a single-qubit operation, the operation has to be allocated with 3 qubits. It's kind of unfortunate that we actually don't throw an exception in case you try to use more qubits. However, just as a side note, I expect the `nqubits` argument of the Operation class to go away in the near term future as it is only a technical dept of our decision diagram package.
```suggestion
TEST_F(QFRFunctionality, addControlStandardOperation) {
  auto op = StandardOperation(3, 0, OpType::X);

  op.addControl(1_pc);
  op.addControl(2_pc});

  ASSERT_EQ(op.getNcontrols(), 2);
  const auto expectedControls = Controls{1_pc, 2_pc};
  EXPECT_EQ(op.getControls(), expectedControls);
  op.removeControl(1_pc);
  const auto expectedControlsAfterRemove = Controls{2_pc};
  EXPECT_EQ(op.getControls(), expectedControlsAfterRemove);
  op.clearControls();
  EXPECT_EQ(op.getNcontrols(), 0);
}
```